sycl/source/detail/xpti_registry.hpp

```diff
@@ -31,6 +31,15 @@ inline constexpr const char *SYCL_PIDEBUGCALL_STREAM_NAME = "sycl.pi.debug";
 
 class XPTIRegistry {
 public:
+  void initializeFrameworkOnce() {
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+    if (!MInitialized) {
```

Comment:
Correct. Right now this is only called from a single place, which is guarded `std::call_once`, but the future instrumentations will initialize streams closer to use place, so that we don't notify tools about streams we're not going to use.

Reply:
I have replaced this with std::call_once, it better demonstrates my intentions and is definitely thread safe.